ovs/lib/storagerouter.py

```diff
@@ -1264,6 +1265,128 @@ def get_logfiles(local_storagerouter_guid):
         return logfilename
 
     @staticmethod
+    @ovs_task(name='ovs.storagerouter.create_hprm_config_files')
+    def create_hprm_config_files(storagerouter_guid, local_storagerouter_guid, parameters):
+        """
+        Create the required configuration files to be able to make use of HPRM (aka PRACC)
+        These configuration will be zipped and made available for download
+        :param storagerouter_guid: The guid of the StorageRouter for which a HPRM manager needs to be deployed
+        :type storagerouter_guid: str
+        :param local_storagerouter_guid: The guid of the StorageRouter this call is executed on
+        :type local_storagerouter_guid: str
+        :param parameters: Additional information required for the HPRM configuration files
+        :type parameters: dict
+        :return: Name of the zipfile containing the configuration files
+        :rtype: str
+        """
+        # Validations
+        required_params = {'port': (int, {'min': 1, 'max': 65535}),
+                           'vpool_guid': (str, Toolbox.regex_guid),
+                           'fragment_cache_on_read': (bool, None),
+                           'fragment_cache_on_write': (bool, None)}
+        Toolbox.verify_required_params(actual_params=parameters,
+                                       required_params=required_params)
+        vpool = VPool(parameters['vpool_guid'])
+        config_path = None
+        storagerouter = StorageRouter(storagerouter_guid)
+        local_storagerouter = StorageRouter(local_storagerouter_guid)
+        for sd in vpool.storagedrivers:
+            if sd.storagerouter_guid == storagerouter.guid:
+                if len(sd.alba_proxies) == 0:
+                    raise ValueError('No ALBA proxies configured for vPool {0} on StorageRouter {1}'.format(vpool.name, storagerouter.name))
+                config_path = '/ovs/vpools/{0}/proxies/{1}/config/{{0}}'.format(vpool.guid, sd.alba_proxies[0].guid)
+
+        if config_path is None:
+            raise ValueError('vPool {0} has not been extended to StorageRouter {1}'.format(vpool.name, storagerouter.name))
+
+        alba_backend_name = None
+        fragment_cache_on_read = parameters['fragment_cache_on_read']
+        fragment_cache_on_write = parameters['fragment_cache_on_write']
+        if fragment_cache_on_read is True or fragment_cache_on_write is True:
+            if 'path' in parameters:  # Local caching
+                required_params.update({'path': (str, re.compile('^/$|/.*[^.]$')),
```

Comment:
That doesn't make much sense to me, are you sure this works? I'd expect something like `^/.*[^.]?$`